getopt.c

```diff
@@ -0,0 +1,69 @@
+#define _POSIX_C_SOURCE 200809L
+#include <mrsh/getopt.h>
+#include <stdio.h>
+
+char *mrsh_optarg = NULL;
+int mrsh_optind = 1;
+int mrsh_opterr = 1;
+int mrsh_optopt = 0;
+
+int mrsh_getopt(int argc, char *const argv[], const char *optstring) {
+	mrsh_optarg = NULL;
+
+	if (mrsh_optind >= argc) {
+		return -1;
+	}
+
+	if (argv[mrsh_optind] != NULL) {
```

Comment:
We check this condition here, but if it's NULL we access `argv[mrsh_optind][1]` anyway afterwards (line 38). I don't think we need to catch NULL values here since `argv` isn't supposed to be NULL-terminated?

Reply:
`argv` is NULL terminated:

> The argv and environ arrays are each terminated by a null pointer. The null pointer terminating the argv array is not counted in argc.

From http://pubs.opengroup.org/onlinepubs/9699919799/functions/exec.html which is referenced by the getopt docs.

I'll add something to handle `NULL`.

Reply:
Then `argc` should exactly point to the NULL element? This could be an `assert`.

Reply:
Yes, that's true.